application/templates/_helpers.tpl

```diff
@@ -68,3 +68,18 @@ reference:
   kind: Route
   name: {{ include "application.name" . }}
 {{- end }}
+
+{{- define "application.service-account-name" }}
+{{- if .Values.rbac.enabled }}
+  {{- if and .Values.rbac.serviceAccount.enabled .Values.rbac.existingServiceAccountName }}
+    {{- fail "Conflict: 'rbac.existingServiceAccountName' is set, but a new service account is being created. Please disable 'rbac.serviceAccount.enabled' or unset 'rbac.existingServiceAccountName'." }}
+  {{- end }}
+  {{- if .Values.rbac.serviceAccount.enabled }}
+    {{- default (include "application.name" .) .Values.rbac.serviceAccount.name }}
+  {{- else }}
+    {{- default "null" .Values.rbac.existingServiceAccountName }}
```

Comment:
Edge case, may or may not be something to take into consideration: service accounts named `null`. Also, we're assuming the result of this partial template to be interpreted as YAML rather than a raw string, that may create confusion.

```suggestion
    {{- $saName := .Values.rbac.existingServiceAccountName }}
    {{- empty $saName | ternary (quote $saName) "null" }}
```

Reply:
the partial now only returns strings, do you think i should still quote the output?

Reply:
I think it's a good idea, yes, since the output is supposed to be a string.